src/index.ts

```diff
@@ -111,6 +89,141 @@ const run = async (config: R2Config) => {
     return map;
 };
 
+const uploadMultiPart: UploadHandler<CompleteMultipartUploadCommandOutput> = async (file: string, config: R2Config) => {
+
+    const fileName = file.replace(config.sourceDir, "");
+    const fileKey = path.join(config.destinationDir !== "" ? config.destinationDir : config.sourceDir, fileName);
+    const mimeType = mime.getType(file);
+
+    const createMultiPartParams: CreateMultipartUploadCommandInput = {
+        Bucket: config.bucket,
+        Key: fileKey,
+        ContentType: mimeType ?? 'application/octet-stream'
+    }
+
+    const cmd = new CreateMultipartUploadCommand(createMultiPartParams);
+
+    const created = await S3.send(cmd);
+
+    const chunkSize = 10 * 1024 * 1024; // 10MB
+
+    const multiPartMap = {
+        Parts: []
+    }
+
+    const totalSize = formatFileSize(file);
+    let bytesRead = 0;
+    let partNumber = 0;
+    let interrupted = false;
+    const uploads = [];
+    for await (const chunk of readFixedChunkSize(file, chunkSize)) {
+
+        const uploadPartParams: UploadPartCommandInput = {
+            Bucket: config.bucket,
+            Key: fileKey,
+            PartNumber: ++partNumber,
+            UploadId: created.UploadId,
+            Body: chunk,
+        }
+
+        const uploadPart = async (partNumber: number) => {
+            const cmd = new UploadPartCommand(uploadPartParams);
+            let retries = 0
+            while (retries < config.maxTries) {
+                if (interrupted) {
+                    console.info(`R2 Info - Aborting upload part ${partNumber} of ${file} due to previous error`)
+                    return;
+                }
+                try {
+                    const result = await S3.send(cmd);
+                    multiPartMap.Parts.push({ PartNumber: partNumber, ETag: result.ETag });
+                    break;
+                } catch (err: any) {
+                    retries++;
+                    console.error(`R2 Error - ${err.message}, retrying: ${retries}/${config.maxTries}`, err);
+                    await sleep(300);
+                }
+            }
+            if (retries >= config.maxTries) {
```

Comment:
This doesn't seem to run. I still have pending multipart uploads in my R2 bucket, and I've let the program run for a little bit.

Reply:
I never met upload failed so i didn't verify this, how did u make it error?